addons/hr_evaluation/hr_evaluation.py

```diff
@@ -1,374 +1,456 @@
-            next_date = (now + relativedelta(months=emp.evaluation_plan_id.month_next)).strftime('%Y-%m-%d')
-            self.write(cr, uid, [emp.id], {'evaluation_date': next_date}, context=context)
-            plan_id = obj_evaluation.create(cr, uid, {'employee_id': emp.id, 'plan_id': emp.evaluation_plan_id.id}, context=context)
-            obj_evaluation.button_plan_in_progress(cr, uid, [plan_id], context=context)
-        return True
 
+class calendar_event(models.Model):
 
-class hr_evaluation(osv.Model):
-    _name = "hr_evaluation.evaluation"
-    _inherit = ['mail.thread']
-    _description = "Employee Appraisal"
-    _columns = {
-        'date': fields.date("Appraisal Deadline", required=True, select=True),
-        'employee_id': fields.many2one('hr.employee', "Employee", required=True),
-        'note_summary': fields.text('Appraisal Summary'),
-        'note_action': fields.text('Action Plan', help="If the evaluation does not meet the expectations, you can propose an action plan"),
-        'rating': fields.selection([
-            ('0', 'Significantly below expectations'),
-            ('1', 'Do not meet expectations'),
-            ('2', 'Meet expectations'),
-            ('3', 'Exceeds expectations'),
-            ('4', 'Significantly exceeds expectations'),
-        ], "Appreciation", help="This is the appreciation on which the evaluation is summarized."),
-        'survey_request_ids': fields.one2many('hr.evaluation.interview', 'evaluation_id', 'Appraisal Forms'),
-        'plan_id': fields.many2one('hr_evaluation.plan', 'Plan', required=True),
-        'state': fields.selection([
-            ('draft', 'New'),
-            ('cancel', 'Cancelled'),
-            ('wait', 'Plan In Progress'),
-            ('progress', 'Waiting Appreciation'),
-            ('done', 'Done'),
-        ], 'Status', required=True, readonly=True, track_visibility='onchange', copy=False),
-        'date_close': fields.date('Ending Date', select=True),
-    }
-    _defaults = {
-        'date': lambda *a: (parser.parse(datetime.now().strftime('%Y-%m-%d')) + relativedelta(months=+1)).strftime('%Y-%m-%d'),
-        'state': lambda *a: 'draft',
-    }
-
-    def name_get(self, cr, uid, ids, context=None):
-        if not ids:
-            return []
-        reads = self.browse(cr, uid, ids, context=context)
-        res = []
-        for record in reads:
-            name = record.plan_id.name
-            employee = record.employee_id.name_related
-            res.append((record['id'], name + ' / ' + employee))
+    """ Model for Calendar Event """
+    _inherit = 'calendar.event'
+
+    @api.model
+    def create(self, vals):
+        res = super(calendar_event, self).create(vals)
+        if self.env.context.get('active_model') == 'hr_evaluation.evaluation':
+            evaluation_obj = self.env['hr_evaluation.evaluation'].browse(self.env.context.get('active_id', []))
+            evaluation_obj.log_meeting(res.name, res.start)
```

Comment:
Why calling log_meeting from here, you are performing write operation on hr_evaluation.evaluation, write already calling log_meeting, in write of evaluation object just check if interview_deadline then call log_meeting